### ISSUE TITLE
gold-standard training data

### ISSUE THREAD

**Issue by tarakc02** [ppact/extraction#35]
Based on @baileyb0t's review in #29, we know we need cleaner training labels to make progress on the news classificaton task. we have a couple of leads on how to do this:

- @baileyb0t has already labeled some data using a secondary labeling scheme. let's get that into a spreadsheet and pull it into our import pipeline. We can include the for sure positives and negatives -- if we're not sure about a label, we can keep it out of the training data and instead bring it to next Tuesday's check-in. 
- we can leverage the model @ayyubibrahimi has trained so far. based on bailey's review, the training data still contains a good number of true positives (50% of the sample, I suspect the overall prevalence is *much* lower than that), meaning it's possible the model is able to pick up something useful. ayyub is currently reviewing performance on the held-out test data. if that shows that the model is even a little bit better than chance (e.g. even if accuracy on the test set is like 65-75%, which is not enough for production), it's probably worth using that to flag additional records for review to add to our label import pipeline.

**Comment by ayyubibrahimi** [comment]
The model performs pretty poorly on the test set. Accuracy is around 55%. Cleaner training labels are the way.

**Comment by ayyubibrahimi** [comment]
@baileyb0t here are the 30 relevant and unique articles from the test dataset that I reviewed: [review_testdf.csv](https://github.com/ppact/extraction/files/8734137/review_testdf.csv) as mentioned in #29

**Comment by ayyubibrahimi** [comment]
I think that this point was made, but in case it wasn't: 1) there are articles that reference the same incident in the training dataset, 2) the labels for these articles are not consistent.

**Comment by baileyb0t** [comment]
@ayyubibrahimi Thank you for the csv! My hope is to get all of the currently-relevant articles reviewed and relabeled, so that the new labels can be used in generating train-test data. The import task will be updated to reflect this likely next week.

Regarding the duplicate incidents, hopefully most of that will be caught by the work in #34. The inconsistent labels, however, should've been resolved by the correct_relevant() method in import.py. I suspect if the labels were still inconsistent then it's most likely a result of the label quality issue. Can you share a sample of article_ids you noticed referring to same incident with different labels so I can use them in auditing? If they are part of the 30 in the csv then nvm, I'll see them soon!

**Comment by ayyubibrahimi** [comment]
Awesome! Sorry if I stepped on your workflow with my review. My curiosity got the best of me after the model performed so poorly on the test data, and so I thought I'd share my findings. 

The csv that I attached is comprised of articles labeled relevant. Happy to provide a new csv with examples of duplicate articles/conflicting labels.

**Comment by ayyubibrahimi** [comment]
Just read through all articles in the test dataset. Confirmed that I was mistaken in believing that it contained duplicates with inconsistent labels. 

On another note: some of these articles are extremely violent. I can imagine that it will have a jarring effect on the user experience if there is no disclaimer. Any thoughts about how to ameliorate this potential issue?

**Comment by baileyb0t** [comment]
I'm not familiar with the frontend of this project so this might not be helpful, but I think the simplest solution would be to add a warning or disclaimer on the user end before any content would be shown, something the user has to interact with in order for it to go away. Alternatively, we could treat the relevant column as a content warning flag and issue a cw before any of these articles are displayed if contents are shown in batches or individually.

**Comment by tarakc02** [comment]
great stuff, thanks! are these incorporated into `train.parquet` yet? if not, let us know when they are -- as discussed before, i think we can include all of these in train.parquet (treating "possibly false" as false), and then continue to label in order to build up a decent test set. 76/184 is still a reasonable proportion of positives for our training data. I do not think it's that important to have class balance in the test set, as long as we have a reasonable number of positives (and negatives) -- we can vary our evaluation metrics on the test data so we're not just measuring accuracy but including F1, ROC, precision + recall...

**Comment by ayyubibrahimi** [comment]
I created an account for you @baileyb0t in case you're interested in how the articles will be presented. 

url: llead.co (news articles will be availble on the front page); https://llead.co/dept/new-orleans-pd/ (and department pages)
username: bailey@hrdag.org
password: set it [here](https://llead.co/forgot-password/)

**Comment by baileyb0t** [comment]
@ayyubibrahimi Thank you for adding me, I was able to sign in and poke around and this looks excellent!

@tarakc02 I was going to wait to pull into train-test sets until I finished relabeling but I see your point, I'll get a version of the import working so that it pulls in these updates and generates the new training set and confirm when it's ready to pull.

**Comment by baileyb0t** [comment]
@ayyubibrahimi Updated import task available to pull!

- (Mostly a reminder to myself) Because train and test sets are created at the same time, in order to update the training data with the new labels so that most of the data went to training per @tarakc02, I set `train_size=0.99, test_size=0.01`. This should be reverted to 80/20 when the relabeling is done!
- There is an issue with how positive/negative cases (or the features that determine them) are generated that enables overlap. Most of the conflict stems from the updated labels and how the idx creation methods check officer_id and not relevant, so amending those methods to use the new relevant data resolved most of the overlap. However, it's still happening, so I need to investigate this further.

**Comment by ayyubibrahimi** [comment]
Awesome. Thank you! I'll let you know how it goes.

**Comment by ayyubibrahimi** [comment]
Successful pull! Thanks again

**Comment by baileyb0t** [comment]
I fixed the pos/neg classification error, new pull available!

**Comment by ayyubibrahimi** [comment]
Success. Thanks!

**Comment by ayyubibrahimi** [comment]
@baileyb0t just uploaded an updated to_label.xlsx table with all new labels; [link](https://www.dropbox.com/scl/fi/hbbzzeq301njxqepibfq6/to_label_ai.xlsx?dl=0&rlkey=k3hgep2lm0wk1e62q251ejq1l).
